src/shared/DMIParser.c

```diff
@@ -0,0 +1,100 @@
+/*
+ * Created by Aavild on 19/11/2021.
+ */
+#include <stdlib.h>
+#include <stdio.h>
+#include <string.h>
+#include <cJSON.h>
+#include "DMIParser.h"
+/*#include <curl/curl.h>*/
+
+/*input ninjo2dnudk.json and you will get out an array of windValues and a timestamp for last update*/
+windValue* ParseStringToWind(const char *input, int *lastUpdate)
+{
+    cJSON *json = cJSON_Parse(input);
+    if (json == NULL)
+    {
+        const char *error = cJSON_GetErrorPtr();
+        if (error != NULL)
+        {
+            printf("Error at: %s", error);
+        }
+        return 0;
+    }
+
+    cJSON *lastUpdateJson = cJSON_GetObjectItem(json, "lastupdate");
+    if (cJSON_IsString(lastUpdateJson) && (lastUpdateJson->valueint != 0))
+    {
+        printf("Input file last updated: \"%i\"\n", lastUpdateJson->valueint);
+    }
+    /*Test hertil i første omgang*/
+    windValue *values = malloc(sizeof(windValue)*95); /*timeserie of size 95*/
+    if(values == NULL)
+        return 0;  /* out of memory! */
+    int iterator = 0;
+    cJSON *timeserie = cJSON_GetObjectItem(json, "timeserie");
+    cJSON *hour;
+    cJSON_ArrayForEach(hour, timeserie)
+    {
+        cJSON *time = cJSON_GetObjectItem(hour, "time");
+        cJSON *windSpeed = cJSON_GetObjectItem(hour, "windSpeed");
+
+        if (!cJSON_IsNumber(time) || !cJSON_IsNumber(windSpeed))
+        {
+            continue;
+        }
+        values[iterator].timestamp = time->valueint;
+        values[iterator].windspeed = hour->valuedouble;
+        iterator++;
+    }
+
+    cJSON_Delete(json);
+    return values;
+}
+
+/*cURL*/
+/*static size_t cb(void *data, size_t size, size_t nmemb, void *userp)
```

Comment:
Please remove code code. You can always save this on your own system in case we need it for later. But pushing zombie code to the main branch is unacceptable.

Reply:
resolved